File: recursive_max.py
## using recursion find the max item in a list

def recursive_max(a_list):
    if len(a_list) == 0:
        return None
    elif len(a_list) == 1:
        return a_list[0]
    elif len(a_list) == 2:
        return a_list[0] if a_list[0] > a_list[1] else a_list[1]
    else:
        temp_max = recursive_max(a_list[1:])
        return a_list[0] if a_list[0] > temp_max else temp_max


print(recursive_max([1,10,3,4,5,6,7,8,9]))
print(recursive_max([0,100]))
print(recursive_max([]))
print(recursive_max([4]))

File: recursive_max.js
// recursive max of a list

function recursiveMax(anArray) {
    if (anArray.length === 0 ) return null;
    else if (anArray.length === 1 ) return anArray[0];
    else if (anArray.length === 2){
        return (anArray[0] > anArray[1]) ? anArray[0] : anArray[1];
    }
    else {
        let temp_max = recursiveMax(anArray.slice(1));
        return (anArray[0] > temp_max) ? anArray[0] : temp_max;
    }
}

console.log(recursiveMax([1,2,3,5,3,2]));
console.log(recursiveMax([1,2]));
console.log(recursiveMax([0]));
console.log(recursiveMax([]));